src/Message/UserMessage.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Message;
+
+use PhpLlm\LlmChain\Message\Content\ImageUrlContent;
+use PhpLlm\LlmChain\Message\Content\TextContent;
+
+final readonly class UserMessage implements MessageInterface
+{
+    /**
+     * @var list<ImageUrlContent|string>
+     */
+    public array $images;
+
+    public function __construct(
+        public TextContent|string $content,
+        ImageUrlContent|string ...$images,
+    ) {
```

Comment:
i feel like this is a bit limited in comparison to the contract in the end which is just a list of content entries.

that's why i proposed `Message::ofUser(string|Content ...$content)` in the first place.
my thought was it's an array of `Content` but it would be nice to enable just a string as well.

so the logic could be the Message service method does an upcasting of a `string` entries of that `$content` array to `Text` instances, since this is the most prominent use-case. `Image` instances would be properly instantiated by the user themselves.

WDYT?

Reply:
Aaaah! Yeah ... ok ... thats fine! Will do it and adjust the tests. Had the upcasting now in the UserMessage Object but yeah, moving it a bit forward. Would then look like the following with images.

```
// Text as String
Message::ofUser('foo', new Image('http://bar'), new Image ('http://baz'));
// Result: Text, Image, Image

# Text as Content
Message::ofUser(new Text('foo'), new Image('http://bar'), new Image ('http://baz'));
// Result: Text, Image, Image

# All as String
Message::ofUser('foo', 'http://bar', 'http://baz');
// Result: Text, Image, Image

# Only first string can be text ... all others has to be images
Message::ofUser('foo', 'http://bar', new Image ('http://baz'));
// Result: Text, Image, Image
```

What do you think? A lot of casting but looks something consistent?

Reply:
Yeah, thought about the http prefix upcasting for Images but quite skeptical if safe to assume it's always an Image. I know for example that audio support was planned and could Imagine URL Support for that as well.
But we could just break the contract later on as well 👍

Reply:
Checking for a valid HTTP URL would not work. You could also embbed images. That is what i personally do. So base64 encoded images are always valid as a string. So the check would have to also check for data urls 🤔

Reply:
let's keep it simple for now. only upcast is `string` => `new Text` - if you want an image use `new Image`

fair?

Reply:
😯  ... Ähm ... yeah .. simple ... already wrote the bit more complex code with tests. If you want to keep it simple i am open to remove the complexity from the code. Should i? 😉

Reply:
So... the answer was yes: https://github.com/php-llm/llm-chain/pull/25#discussion_r1773830195 😜

Reply:
sorry, didn't catch that in time